Sources/PayoneerCheckout/PaymentProcess/RequestSender.swift

```diff
@@ -172,12 +176,37 @@ private struct PaymentRequestBuilder: Loggable {
         return operationRequest
     }
 
+    private func createDictionary(forExtraElementsIn sections: Set<Input.Network.UIModel.InputSection>) -> [String: Bool]? {
+        // Extract only extra element input fields
+        let checkboxInputFields = [
+            sections[.extraElements(at: .top)]?.inputFields,
+            sections[.extraElements(at: .bottom)]?.inputFields
+        ].compactMap { $0 }.flatMap { $0 }
+
+        // Transform input fields to dictionary with name / value
+        var dictionary = [String: Bool]()
+        for inputField in checkboxInputFields {
+            // Work only with checkboxes, extra elements could containt other items like Labels
+            guard let checkbox = inputField as? Input.Field.Checkbox else { continue }
+
+            // Get id as a `String`
+            guard case .extraElement(let name) = inputField.id else { continue }
+
+            dictionary[name] = checkbox.isOn
+        }
```

Comment:
Could use a `reduce` here:

```suggestion
        let dictionary = checkboxInputFields.reduce(into: [String: Bool]()) { partialResult, inputField in
            guard let checkbox = inputField as? Input.Field.Checkbox else { return }
            guard case .extraElement(let name) = inputField.id else { return }
            partialResult[name] = checkbox.isOn
        }
```